e2e/driver/driver.js

```diff
@@ -1,107 +1,157 @@
 // @flow
 
-import childProcess from 'child_process'
 import wd from 'wd'
-import serverConfigs from '../config/server-configs'
-import caps from '../config/caps'
+import fetch from 'node-fetch'
+import childProcess from 'child_process'
+import serverConfigs from '../config/configs'
 import { clone } from 'lodash'
 
-const defaultPlatform = 'android'
-const platform = process.env.E2E_PLATFORM || defaultPlatform
-
-const defaultServer = 'local'
-const e2eServerConfigName = process.env.E2E_SERVER || defaultServer
-
-const defaultCaps = 'local_android9'
-const capsName = process.env.E2E_CAPS || defaultCaps
-
 const BROWSERSTACK_EXHAUSTED_MESSAGE = 'All parallel tests are currently in use, including the queued tests. ' +
   'Please wait to finish or upgrade your plan to add more sessions.'
 const IMPLICIT_WAIT_TIMEOUT = 80000
 const INIT_RETRY_TIME = 3000
 const STARTUP_DELAY = 3000
 
+const ADDITIONAL_ENV_VARIABLES = ['CIRCLE_BUILD_URL', 'CIRCLE_BUILD_NUM']
+
 export const timer = (ms: number) => new Promise<{}>(resolve => setTimeout(resolve, ms))
 
-export const isAndroid = () => {
-  return platform.toLowerCase() === 'android'
+type ConfigType = { url: string, platform: string, prefix: string, caps: {} }
+
+const getConfig = (): ConfigType | null => {
+  const configName: ?string = process.env.E2E_CONFIG
+
+  if (!configName) {
+    console.error('E2E_CONFIG name is not set!')
+    return null
+  }
+
+  const config = serverConfigs[configName.toLowerCase()]
+
+  if (!config) {
+    console.error(`Server config ${configName} not found!`)
+    return null
+  }
+
+  return config
+}
+
+export const isAndroid = (config: ConfigType) => {
+  return config.platform.toLowerCase() === 'android'
 }
 
-export const isIOS = () => {
-  return platform.toLowerCase() === 'ios'
+export const isIOS = (config: ConfigType) => {
+  return config.platform.toLowerCase() === 'ios'
+}
+
+const getAdditionalTags = () => {
+  return ADDITIONAL_ENV_VARIABLES.map(variable => process.env[variable]).filter(value => !!value)
 }
 
 export const select = <T, K> (input: { android: T, ios: K }): T | K => {
```

Comment:
Not quite sure if it wouldn't make more sense to just inline this in `setupDriver`. But since it works right now, it's probably fine the way it is. Just took me a while to understand why it's used this way since it's a little bit complicated.